### PR TITLE
fix: secret lock for multi-user scenario

### LINKED ISSUE #101: Update "master" key term

### LINKED ISSUE #110: Fix issue with creating secret lock service
Investigate and fix crash in AuthZ KMS when "secrets split lock" is used for multi-user scenario:

```
ERROR Failed to create a KMS service: failed to create new secret lock service: cipher: message authentication failed
```

### PR DESCRIPTION
**Pull request by aholovko**
Fixes #110
Closes #101

Signed-off-by: Andriy Holovko <andriy.holovko@gmail.com>

### CODE REVIEW

**Review by llorllale — APPROVED** [review]

File: pkg/secretlock/service.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+func keyEntryInDB(keyURI string) string {
+	return strings.ReplaceAll(keyURI, localLockPrefix, "")
+}
+
+func randomBytes(size uint32) []byte {
+	buf := make([]byte, size)
+
+	_, err := rand.Read(buf)
+	if err != nil {
+		panic(err)
```
Comment (llorllale) [inline pkg/secretlock/service.go:99]: panic?

**Review by rolsonquadras — APPROVED** [review]

File: pkg/secretlock/service.go (hunk 1 above)
Comment (llorllale) [reply pkg/secretlock/service.go:99]: should propagate this error back to `New()`

Comment (aholovko) [reply pkg/secretlock/service.go:99]: out of randomness, should never happens :)
but never say never so what is your suggestion, should we just log that issue?

Comment (llorllale) [reply pkg/secretlock/service.go:99]: @aholovko to propagate it all the way up to `New()` and return it

Comment (aholovko) [reply pkg/secretlock/service.go]: done

**Pull request merged**